processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java

```diff
@@ -100,6 +98,29 @@ public static void throwIfInvalidProtocols(HttpInputSourceConfig config, List<UR
     }
   }
 
+  public static void throwIfForbiddenHeaders(HttpInputSourceConfig config, Map<String, String> headersMap)
+  {
+    if (!config.getAllowedHeaders().isEmpty() && headersMap.size() > 0) {
+      Set<String> forbiddenHeaderSet = headersMap.keySet()
+                                                 .stream()
+                                                 .map(StringUtils::toLowerCase)
+                                                 .filter(h -> !config.getAllowedHeaders().contains(h))
+                                                 .collect(Collectors.toSet());
+      if (!forbiddenHeaderSet.isEmpty()) {
+        throw new IAE("Got forbidden headers %s, allowed headers are only %s ",
+                      forbiddenHeaderSet, config.getAllowedHeaders());
+      }
+    }
```

Comment:
this could be simplified to 
```
for key in headersMap
  if (!config.allowedHeaders.contains(key))
     throw new IAE(" Header [%s] is not allowed to be set. Only headers are allowed are [%s]. You can allow the headers by changing property <insert property name> ",
                      key, config.getAllowedHeaders());
```

Reply:
Also please use InvalidInput.exception.

Reply:
fixed